src/sinks/elasticsearch/encoder.rs

```diff
@@ -70,18 +72,29 @@ fn write_bulk_action(
     bulk_action: &str,
     index: &str,
     doc_type: &str,
+    suppress_type: bool,
     id: &Option<String>,
 ) -> std::io::Result<usize> {
     as_tracked_write(
         writer,
-        (bulk_action, index, doc_type, id),
-        |writer, (bulk_action, index, doc_type, id)| {
+        (bulk_action, index, doc_type, id, suppress_type),
+        |writer, (bulk_action, index, doc_type, id, suppress_type)| {
```

Comment:
Nit: I personally find this sort of boolean table more readable as a match statement:

```
match (id, supress_type) {
  Some(id), true => {},
  Some(id), false => {}
  None, true => {}
  None, false => {}
}
```

But not a big deal.

Reply:
I also find it super unreadable 👍